drivers/soundwire/intel_init.c

```diff
@@ -69,23 +69,13 @@ static int sdw_intel_cleanup(struct sdw_intel_ctx *ctx)
 		if (link_mask && !(link_mask & BIT(i)))
 			continue;
 
-		if (!IS_ERR_OR_NULL(link->md)) {
-			ret = sdw_master_device_del(link->md);
-			if (ret < 0)
-				dev_err(&link->md->dev,
-					"master device del failed %d\n",
-					ret);
-		}
+		if (!link->bus)
```

Comment:
This should be ``if (link->bus)`` and it was the reason of the sof_remove.sh regression.

Reply:
ok will fix, thanks for spotting this!